### PR TITLE
fix(cli): explicitly scan for ignore attribute in inline tests

### PR DESCRIPTION
**Pull request by caspervonb**
This adds `ignore` as a known attribute which drops a code block early whenever it is seen in documentation tests.

See https://github.com/denoland/deno/issues/11645

### CODE REVIEW

File: cli/tests/test/doc_markdown/doc.md [hunk 1]
```diff
@@ -18,6 +18,12 @@ The following block should be given a ts extension on extraction:
 console.log("ts");
 ```
 
+The following example contains the ignore attribute and will be ignored:
+
+```ts ignore
```
[the remark this answers is not in the document]
Comment (caspervonb) [reply cli/tests/test/doc_markdown/doc.md:23]: > See here:
> 
> Input: https://raw.githubusercontent.com/dsherret/markdown-test/main/test.md
> Output: https://dsherret.github.io/markdown-test/test

Aha, thanks that makes it painfully obvious.
That's annoying but I'm not sure we can do anything better without getting awkward.

Prior art here in rustdoc's `ignore` attribute as-well so as long as most CommonMark and GFM renderers get it right, do we care?

**Comment by bartlomieju** [comment]
@dsherret what do you think? Can we land it?

**Comment by caspervonb** [comment]
> @bartlomieju I've been thinking about this a bit more now (regarding discussion in [#11647 (comment)](https://github.com/denoland/deno/pull/11647#discussion_r686831881)) and I think it's desirable for it to work this way. Perhaps in the future we can add additional support for making this work with non-GFM processors (ex. with a comment).
> 
> @caspervonb one thing I'm not sure about though is that "ignore" is very general. Perhaps it should be more specific to avoid conflicts? `deno-test-ignore`?

I'm not too worried about it and would prefer to stick with the brief readable version than add to the pile of long directives.
There's prior art for it in rustdoc for this as-well which is the tool we want to be like when this is all wrapped up.

The only thing we're really committing to here is that any fenced code block with `ignore` in the info-string is ignored regardless of what other things the info string contains.

The info-string `ts ignore` happens to render nicely for us now as an escape hatch for getting the mostly manual type-checked but we'll provide better ways to do this with `no_run` and `no_check` soon enough when the blocks are runnable.

**Review by bartlomieju — APPROVED** [review]
LGTM too, let's go roll with `ignore` for now and bike-shed on different attribute later.

**Pull request merged**